src/client/src/StreamState.c

```diff
@@ -9,36 +9,140 @@
+        fromGetEndpointStreamState,
+        executeGetEndpointStreamState,
+        defaultStreamStateMachineErrorHandler,
+        SERVICE_CALL_MAX_RETRY_COUNT,
+        STATUS_GET_STREAMING_ENDPOINT_CALL_FAILED
+    },
+    {
+        STREAM_STATE_GET_TOKEN,
+        STREAM_STATE_STOPPED | STREAM_STATE_GET_ENDPOINT | STREAM_STATE_GET_TOKEN,
+        fromGetTokenStreamState,
+        executeGetTokenStreamState,
+        defaultStreamStateMachineErrorHandler,
+        SERVICE_CALL_MAX_RETRY_COUNT,
+        STATUS_GET_STREAMING_TOKEN_CALL_FAILED
+    },
+    {
+        STREAM_STATE_READY,
+        STREAM_STATE_STOPPED | STREAM_STATE_GET_TOKEN | STREAM_STATE_READY | STREAM_STATE_PUT_STREAM | STREAM_STATE_STREAMING,
+        fromReadyStreamState,
+        executeReadyStreamState,
+        defaultStreamStateMachineErrorHandler,
+        SERVICE_CALL_MAX_RETRY_COUNT,
+        STATUS_STREAM_READY_CALLBACK_FAILED
+    },
+    {
+        STREAM_STATE_PUT_STREAM,
+        STREAM_STATE_STOPPED | STREAM_STATE_READY | STREAM_STATE_PUT_STREAM,
+        fromPutStreamState,
+        executePutStreamState,
+        defaultStreamStateMachineErrorHandler,
+        INFINITE_RETRY_COUNT_SENTINEL,
+        STATUS_PUT_STREAM_CALL_FAILED
+    },
+    {
+        STREAM_STATE_STREAMING,
+        STREAM_STATE_STOPPED | STREAM_STATE_PUT_STREAM | STREAM_STATE_STREAMING,
+        fromStreamingStreamState,
+        executeStreamingStreamState,
+        defaultStreamStateMachineErrorHandler,
+        INFINITE_RETRY_COUNT_SENTINEL,
+        STATUS_PUT_STREAM_CALL_FAILED
+    },
+    {
+        STREAM_STATE_STOPPED,
+        STREAM_STATE_STOPPED | STREAM_STATE_CREATE | STREAM_STATE_DESCRIBE | STREAM_STATE_TAG_STREAM | STREAM_STATE_GET_ENDPOINT |
+        STREAM_STATE_GET_TOKEN | STREAM_STATE_READY | STREAM_STATE_PUT_STREAM | STREAM_STATE_STREAMING,
+        fromStoppedStreamState,
+        executeStoppedStreamState,
+        defaultStreamStateMachineErrorHandler,
+        INFINITE_RETRY_COUNT_SENTINEL,
+        STATUS_PUT_STREAM_CALL_FAILED
+    }
 };
 
 UINT32 STREAM_STATE_MACHINE_STATE_COUNT = SIZEOF(STREAM_STATE_MACHINE_STATES) / SIZEOF(StateMachineState);
 
 ///////////////////////////////////////////////////////////////////////////
 // State machine callback functions
 ///////////////////////////////////////////////////////////////////////////
+
+
+STATUS defaultStreamStateMachineErrorHandler(UINT64 customData /* customData should be PKinesisVideoStream */) {
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PKinesisVideoStream pKinesisVideoStream = NULL;
+    PKinesisVideoClient pKinesisVideoClient = NULL;
+    PKvsRetryStrategy pKvsRetryStrategy = NULL;
+
+    pKinesisVideoStream = STREAM_FROM_CUSTOM_DATA(customData);
+    CHK(pKinesisVideoStream != NULL, STATUS_NULL_ARG);
+
+    pKinesisVideoClient = pKinesisVideoStream->pKinesisVideoClient;
+    CHK(pKinesisVideoStream != NULL, STATUS_NULL_ARG);
```

Comment:
You meant to CHK the client is not null since you already checked the stream above before getting the client out of it.